File: admin/web/src/api/ddz/robotConfig.js
import adminService from '@/utils/adminRequest'

// 获取机器人配置列表
export function getRobotConfigList(data) {
  return adminService({
    url: '/ddz/robotConfig/list',
    method: 'post',
    data
  })
}

// 获取机器人配置详情
export function getRobotConfigInfo(id) {
  return adminService({
    url: '/ddz/robotConfig/info',
    method: 'get',
    params: { id }
  })
}

// 获取默认配置
export function getDefaultConfig() {
  return adminService({
    url: '/ddz/robotConfig/default',
    method: 'get'
  })
}

// 获取所有启用的配置（用于下拉选择）
export function getAllConfigs() {
  return adminService({
    url: '/ddz/robotConfig/all',
    method: 'get'
  })
}

// 创建机器人配置
export function createRobotConfig(data) {
  return adminService({
    url: '/ddz/robotConfig/create',
    method: 'post',
    data
  })
}

// 更新机器人配置
export function updateRobotConfig(data) {
  return adminService({
    url: '/ddz/robotConfig/update',
    method: 'put',
    data
  })
}

// 删除机器人配置
export function deleteRobotConfig(id) {
  return adminService({
    url: '/ddz/robotConfig/delete',
    method: 'delete',
    data: { id }
  })
}

// 设置默认配置
export function setDefaultConfig(id) {
  return adminService({
    url: '/ddz/robotConfig/setDefault',
    method: 'post',
    data: { id }
  })
}

File: admin/server/router/ddz/robot_config.go
package ddz

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type DDZRobotConfigRouter struct{}

func (r *DDZRobotConfigRouter) InitDDZRobotConfigRouter(Router *gin.RouterGroup) {
	ddzRobotConfigRouter := Router.Group("ddz/robotConfig").Use(middleware.OperationRecord())
	ddzRobotConfigRouterWithoutRecord := Router.Group("ddz/robotConfig")

	ddzRobotConfigApi := v1.ApiGroupApp.DDZApiGroup.DDZRobotConfigApi
	{
		// 配置管理（需要操作记录）
		ddzRobotConfigRouter.POST("create", ddzRobotConfigApi.CreateRobotConfig)      // 创建配置
		ddzRobotConfigRouter.PUT("update", ddzRobotConfigApi.UpdateRobotConfig)       // 更新配置
		ddzRobotConfigRouter.DELETE("delete", ddzRobotConfigApi.DeleteRobotConfig)    // 删除配置
		ddzRobotConfigRouter.POST("setDefault", ddzRobotConfigApi.SetDefaultConfig)   // 设置默认配置
	}
	{
		// 配置查询（不需要操作记录）
		ddzRobotConfigRouterWithoutRecord.POST("list", ddzRobotConfigApi.GetRobotConfigList)    // 获取配置列表
		ddzRobotConfigRouterWithoutRecord.GET("info", ddzRobotConfigApi.GetRobotConfigByID)     // 获取配置详情
		ddzRobotConfigRouterWithoutRecord.GET("default", ddzRobotConfigApi.GetDefaultConfig)    // 获取默认配置
		ddzRobotConfigRouterWithoutRecord.GET("all", ddzRobotConfigApi.GetAllConfigs)           // 获取所有启用的配置
	}
}
